app/models/index-card.ts

```diff
@@ -90,32 +90,29 @@ export default class IndexCardModel extends Model {
     async getOsfModel(options?: object) {
         const identifier = this.resourceIdentifier;
         if (identifier && this.osfModelType) {
-            const guid = this.guidFromIdentifierList;
+            const guid = this.osfGuid;
             if (guid) {
                 const osfModel = await this.store.findRecord(this.osfModelType, guid, options);
                 this.osfModel = osfModel;
             }
         }
     }
 
-    get osfUrl() {
-        const guid = this.guidFromIdentifierList;
-        if (guid) {
-            return `${osfUrl}${guid}/`;
+    get osfIdentifier() {
+        for (const iri of this.resourceIdentifier) {
+            if (iri && iri.startsWith(osfUrl)) {
+                return iri;
+            }
         }
         return '';
```

Comment:
`.find()` can make this clearer
```suggestion
        return this.resourceIdentifier.find(iri => iri.startsWith(osfUrl)) || '';
```